target/scripts/startup/setup-stack.sh

```diff
@@ -960,6 +960,22 @@ function _setup_security_stack
       sed -i "s|\$final_spam_destiny.*=.*$|\$final_spam_destiny = D_BOUNCE;|g" /etc/amavis/conf.d/49-docker-mailserver
       sed -i "s|\$final_bad_header_destiny.*=.*$|\$final_bad_header_destiny = D_BOUNCE;|g" /etc/amavis/conf.d/49-docker-mailserver
     fi
+
+    if [[ ${ENABLE_SPAMASSASSIN_KAM} -eq 1 ]]
+    then
+      _notify 'inf' 'Configuring Spamassassin KAM'
+      local SPAMASSASSIN_KAM_CRON_FILE=/etc/cron.daily/10-spamassassin_kam
+
+      sa-update --import /etc/spamassassin/kam/kam.sa-channels.mcgrail.com.key
+      cat >"${SPAMASSASSIN_KAM_CRON_FILE}" <<"EOM"
+#! /bin/bash
+
+sa-update --gpgkey 24C063D8 --channel kam.sa-channels.mcgrail.com -v 2>&1
+
+EOM
+
+      chmod +x "${SPAMASSASSIN_KAM_CRON_FILE}"
```

Comment:
This shouldn't be required either if the file is committed with `+x` set, or via Dockerfile build as extra precaution? (_not sure how reliable `+x` is in if building image on platform like windows_)

Reply:
@casperklein made me aware in one of my last PRs that this was missing, so I added it here not sure if it was necessary. I don't see any harm here though, and because we do not support Windows, we do not need to care about whether `chmod +x` work in Windows, right?

Reply:
> because we do not support Windows, we do not need to care about whether `chmod +x` work in Windows, right?

I was referring to not being sure if the `+x` metadata is retained if you checkout/clone from git onto a Windows machine and build the Dockerfile from that.

Which was regarding if this `chmod` line was dropped because we could commit the executable `+x` with a separate `.sh` file. Alternatively, that separate `.sh` file could get a more explicit `chmod +x` line added to the Dockerfile which would resolve any edge case concerns about that issue :sweat_smile: 

---

I'm not fussed either way, but from a maintenance viewpoint with the other approaches to handing cron files elsewhere in `setup-stack.sh`, I don't quite get why the PR does it this way (heredoc?).

If a future PR extracts out related cron specific logic, perhaps the `echo` statements will be converted to this approach with heredoc as well if that seems more appropriate?

Reply:
> Which was regarding if this chmod line was dropped because we could commit the executable +x with a separate .sh file. Alternatively, that separate .sh file could get a more explicit chmod +x line added to the Dockerfile which would resolve any edge case concerns about that issue

Can you elaborate on this again, because I don't get it :) What do you mean by "separate .sh file"?

---

I like the heredoc because you can immediate see the content of the file. With `echo` you need `\n` etc. which I find to not be suitable. In the future PR, I will tend to this issue and make all cron additions heredocs.

Reply:
> Can you elaborate on this again, because I don't get it :) What do you mean by "separate .sh file"?

I think he meant something like [this](https://github.com/docker-mailserver/docker-mailserver/blob/master/target/dovecot/dovecot-purge.cron).

Reply:
Ah, now I see. I actually like the heredoc and `chmod +x` better, but if a majority votes for the extra-script-approach, I'd implement it.

Reply:
I think both attempts are valid and have their pro/cons. But as always: We should keep it uniform 😆

Reply:
Well, at least this is something a future PR will solve :D